snuba/migrations/autogeneration/main.py

```diff
@@ -0,0 +1,30 @@
+import os
+import subprocess
+
+import requests
+import yaml
+
+
+def generate(local_storage_path: str) -> tuple[str, str]:
+    local_repo_path = (
+        subprocess.run(["git", "rev-parse", "--show-toplevel"], capture_output=True)
+        .stdout.decode("utf-8")
+        .strip()
+    )
+    local_storage_path = os.path.abspath(os.path.expanduser(local_storage_path))
+    if not local_storage_path.startswith(local_repo_path):
+        raise ValueError(
+            f"Storage path '{local_storage_path}' is not in the git repository '{local_repo_path}'"
+        )
+
+    # get local storage
+    rel_storage_path = os.path.relpath(local_storage_path, local_repo_path)
+    with open(os.path.join(local_repo_path, rel_storage_path), "r") as f:
+        new_storage = yaml.safe_load(f)
+
+    # get remote storage
+    REMOTE_SNUBA_REPO = "https://raw.githubusercontent.com/getsentry/snuba/master"
+    res = requests.get(f"{REMOTE_SNUBA_REPO}/{rel_storage_path}")
+    old_storage = yaml.safe_load(res.text)
```

Comment:
Have you considered removing this remote git call and comparing whatever is unstaged with the local HEAD? This should be similar to your `subprocess.run` on line 10, with some slightly different arguments.

Some downsides of this approach, in order of descending concern to me:
- hard-codes a base branch, master, which might not be great when working on a new feature
- requires Github API to be up and for the user to have an active internet connection to run the command
- assumes Github API route remains the same over time, and doesn't require authentication
- assumes that we will always consider Github the canonical source of truth (unlikely to change, but I think the less we hard-code this dependency the better)

Reply:
I considered using the local repo instead of remote. The reason I ended up deciding on remote was because I wanted "old storage" to be as close to what actually in clickhouse prod as possible. If users have their own local branch in some specific state this wouldnt be the case.

Reply:
If we are using master it might get confusing if someone's local branch is not up to date. If someone else merged in a migration that you hadn't run yet, we wouldn't want your local migration to include those schema changes, right?

Reply:
@MeredithAnya ok yeah I see what you're saying now... so do you agree with Oliver that comparing whatever is unstaged with the local HEAD might be a better approach?